classes/local/notification/start.php

```diff
@@ -85,4 +85,17 @@ public static function notify_users(?stdClass $program, ?stdClass $user): void {
         }
         $rs->close();
     }
+
+    /**
+     * Send notification related to start.
+     *
+     * @param stdClass $user
+     * @param stdClass $program
+     * @param stdClass $source
+     * @param stdClass $allocation
+     * @return void
```

Comment:
https://moodledev.io/general/development/policies/codingstyle#documentation-and-comments

Void should only be used for explicit empty returns, rather than implicit. In this case, just drop the @return and the type hint in the signature

Reply:
Those should be removed now.